spec/bolt_server/app_integration_spec.rb

```diff
@@ -171,6 +176,191 @@ def app
       end
     end
 
+    describe 'apply_prep' do
+      let(:path) { "/ssh/apply_prep" }
+
+      it 'apply_prep runs install task configured in plugin_hooks and gathers custom facts' do
+        # Target a spec container that already has an agent on it
+        inventory = Bolt::Inventory.empty
+        puppet_6_agent_container = inventory.get_target(conn_uri('ssh', include_password: true, override_port: 20024))
+        target = {
+          hostname: puppet_6_agent_container.host,
+          user: puppet_6_agent_container.user,
+          password: puppet_6_agent_container.password,
+          port: puppet_6_agent_container.port,
+          plugin_hooks: { 'puppet_library' => { 'plugin' => 'task', 'task' => 'fake_puppet_agent::install',
+                                                'parameters' => {} } }
+        }
+        body = {
+          versioned_project: 'bolt_server_test_project',
+          target: target,
+          job_id: 1
+        }
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
+        expect(last_response).to be_ok
+        expect(last_response.status).to eq(200)
+        result = JSON.parse(last_response.body)
+        expect(result['status']).to eq('success')
+        expect(result['value'].keys).to include('os')
+      end
+
+      it 'apply_prep fails when install task fails' do
```

Comment:
Should probably add an "apply_prep fails when custom facts task" fails test as well if we are including this here.

Reply:
I"m not sure that is worth it. The reason I added that test is because we need to ensure the correct behavior (we do not try to run the facts task when the install task fails). Just asserting the facts task fails (which would not really be trivial to do) seems unnecessary.